src/widget/sidebar_filler.c

```diff
@@ -0,0 +1,258 @@
+#include "sidebar_filler.h"
+
+#include "city/labor.h"
+#include "city/population.h"
+#include "city/ratings.h"
+#include "core/config.h"
+#include "core/lang.h"
+#include "core/string.h"
+#include "game/settings.h"
+#include "graphics/arrow_button.h"
+#include "graphics/graphics.h"
+#include "graphics/lang_text.h"
+#include "graphics/menu.h"
+#include "graphics/panel.h"
+#include "graphics/text.h"
+#include "scenario/criteria.h"
+#include "scenario/property.h"
+
+#define EXTRA_INFO_LINE_SPACE 32
+#define EXTRA_INFO_HEIGHT_GAME_SPEED 64
+#define EXTRA_INFO_HEIGHT_UNEMPLOYMENT 112
+#define EXTRA_INFO_HEIGHT_RATINGS 272
+#define EXTRA_INFO_TOP_PADDING 10
+#define EXTRA_INFO_SPEED_PADDING 26
+#define EXTRA_INFO_UNEMPLOYMENT_TOP_PADDING 20
+#define EXTRA_INFO_UNEMPLOYMENT_BOTTOM_PADDING 22
+
+static void button_game_speed(int is_down, int param2);
+
+static arrow_button arrow_buttons_speed[] = {
+    {11, 30, 17, 24, button_game_speed, 1, 0},
+    {35, 30, 15, 24, button_game_speed, 0, 0},
+};
+
+typedef struct {
+    int value;
+    int target;
+} objective;
+
+static struct {
+    int height;
+    int game_speed;
+    int unemployment_percentage;
+    int unemployment_amount;
+    objective culture;
+    objective prosperity;
+    objective peace;
+    objective favor;
+    objective population;
+} data;
+
+int sidebar_filler_is_enabled(void)
+{
+    return config_get(CONFIG_UI_SIDEBAR_INFO);
+}
+
+int sidebar_filler_get_height(void)
+{
+    return data.height;
+}
+
+int sidebar_filler_handle_mouse(const mouse *m, int x_offset, int y_offset)
+{
+    return arrow_buttons_handle_mouse(m, x_offset, y_offset, arrow_buttons_speed, 2);
+}
+
+static int calculate_extra_info_height(int y_offset, int is_collapsed, int sidebar_height)
+{
+    if (is_collapsed || !config_get(CONFIG_UI_SIDEBAR_INFO)) {
+        data.height = 0;
+    } else {
+        int available_height = sidebar_height - y_offset;
```

Comment:
Now you implicitly assume that the sidebar starts at the top of the screen. Just pass `available_height` directly from `sidebar.c` instead of calculating it here.